zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1360,6 +1360,13 @@ public List<String> getChildren(String path) {
   }
 
   protected List<String> getChildren(final String path, final boolean watch) {
+    // Throws exception when try to subscribe watch when using _usePersistWatcher. When ZkClient
+    // is subscribed as persist watcher, resubscribing the same object as onw time watcher will
+    // over write the persist watcher causing missing following event.
+    if (_usePersistWatcher && watch) {
```

Comment:
is this change in API contract? If not, we should be good.

Reply:
TFTR! This is not a public API, it should not change any API contract.